Publishing the Updated TS SDK Material

jmintuitive opened the pull request:
### Description

This includes all updated reference material for the TS SDK.

As of now, I'm 50% through manually reviewing the content. Taking a bit longer than I expected as there are over 100 files with significant documentation changes to read through. 

As the PR is now, it is ready for review but NOT ready to merge. There are 2 final steps needed. 

As you do the PR review, please be sure to mark the files as "read" in the top right corner - that will help you keep track of which ones are actually done as I finish reviewing / committing the other 50 files which have updated docs.

The last step of this PR will be to add the file organization and generate the docs, but I'm waiting on that until after review because that will make this PR impossible to read. 

Please DO NOT MERGE until we have at least 1 approver and I make a comment asking to merge (as that will give me time to make sure everything looks as expected locally - it's a lot of changes! )

### Changes to the docs gen flow (behind the scenes) to get this to work
1. Updating scripts/generateDocs.sh to use a typedoc.json.
2. Downloading a plugin called typedoc-plugin-missing-exports to display things like `Aptos.transaction`'s implementation even though we don't want users to be able to *instantiate* the Transaction class.
3. Setting up `typedoc.json` to use folder structures.

### Tooling to do this going forward
I'll post the tooling and instructions after I get this round of revised content done / ready to go. 

### Test Plan
<!-- Please provide us with clear details for verifying that your changes work. -->
1. **Extensive** manual review.
2. Automated checks (such as verifying that all codeblocks are ```typescript ... ```, ensuring all files end with an extra line, searches for errors, etc.)

### Related Links
<!-- Please link to any relevant issues or pull requests! -->

0xmaayan commented on src/api/aptosConfig.ts:
```diff
@@ -91,14 +135,25 @@ export class AptosConfig {
+   * import { Aptos, AptosConfig, Network, AptosApiType } from "@aptos-labs/ts-sdk";
+   * 
+   * const config = new AptosConfig({ network: Network.TESTNET });
+   * const aptos = new Aptos(config);
```
```suggestion
```

0xmaayan commented on src/api/aptosConfig.ts:
```diff
@@ -128,19 +183,52 @@ export class AptosConfig {
+   * import { Aptos, AptosConfig, Network } from "@aptos-labs/ts-sdk";
+   * 
+   * const config = new AptosConfig({ network: Network.TESTNET });
+   * const aptos = new Aptos(config);
```
```suggestion
```

0xmaayan commented on src/api/aptosConfig.ts:
```diff
@@ -128,19 +183,52 @@ export class AptosConfig {
+   * import { Aptos, AptosConfig, Network } from "@aptos-labs/ts-sdk";
+   *
+   * const config = new AptosConfig({ network: Network.TESTNET });
+   * const aptos = new Aptos(config);
```
```suggestion
```

jmintuitive replied:
Because AptosConfig is always used to create an `Aptos` instance I thought it was worth showing it creating that full class.

0xmaayan commented on src/api/coin.ts:
```diff
@@ -9,27 +9,65 @@ import { AnyNumber, MoveStructId } from "../types";
   /**
-   * Generate a transfer coin transaction that can be simulated and/or signed and submitted
-   *
+   * Initializes a new instance of the Aptos client with the specified configuration.
```
this is not belong here. we are in the Coin class

0xmaayan commented on src/api/event.ts:
```diff
@@ -15,21 +15,63 @@ import { AptosConfig } from "./aptosConfig";
 export class Event {
+
+  /**
+   * Initializes a new instance of the Aptos client with the provided configuration.
```
ohh it looks like you added it to any API class - not completely sure why

jmintuitive replied:
In order for the comments to show up in TS Doc, we have to add comments to the API classes since they're used as mixins to define the functions on `Aptos`. These comments are what actually explain each method :)

0xmaayan reviewed (commented):
There are 174 files to review - it is a lot to handle. In the future, lets break it into a smaller PRs.

jmintuitive commented:
> There are 174 files to review - it is a lot to handle. In the future, lets break it into a smaller PRs.

Yeah, sorry for the large PR. In the future I'll break up PRs like this even though they have the same "purpose" - it's too hard to review in one block

gregnazario commented:
Looks like there's going to be a bunch of conflicts, with keyless.  Probably good to sync with @heliuchuan @jmintuitive

gregnazario commented:
@jmintuitive I've fixed all of the lints, and formatting.  Please fix them next time.

Additionally, I've added some TODOs around some JSDocs.  I also suspect, there were some merge issues, so some of the docs don't make sense... they probably have to be gone over again

jmintuitive commented:
> @jmintuitive I've fixed all of the lints, and formatting. Please fix them next time.
> 
> Additionally, I've added some TODOs around some JSDocs. I also suspect, there were some merge issues, so some of the docs don't make sense... they probably have to be gone over again

Will do today - thank you!
